build.sh

```diff
@@ -26,27 +26,39 @@ MY_DIR="$(dirname $SCRIPT_EXEC)"
 FRAMEWORK_DIR=${MY_DIR}/framework
 UT_CONTROL_LIB_DIR=${FRAMEWORK_DIR}/ut-control
 
+if [[ "$*" == *"VARIANT=CPP"* ]]; then
+    VARIANT="CPP"
+else
+    VARIANT="C"
+fi
+
 if [[ "$*" == *"TARGET=arm"* ]]; then
     TARGET="arm"
 else
     TARGET="linux"
 fi
 echo "TARGET= [$TARGET] from [$0]"
+echo "VARIANT= [$VARIANT] from [$0]"
 
 THIRD_PARTY_LIB_DIR=${FRAMEWORK_DIR}/ut-control/build/${TARGET}
+GTEST_DIR=${FRAMEWORK_DIR}/gtest/${TARGET}
+GTEST_LIB_DIR=${MY_DIR}/build/${TARGET}/cpp_libs
+GTEST_VERSION=1.15.2
 
 pushd ${MY_DIR} > /dev/null
 # Clone CUnit
-if [ ! -d "${FRAMEWORK_DIR}/CUnit-2.1-3" ]; then
+if [[ ! -d "${FRAMEWORK_DIR}/CUnit-2.1-3" && "${VARIANT}" == "C" ]]; then
     echo "Clone Framework"
     wget https://sourceforge.net/projects/cunit/files/CUnit/2.1-3/CUnit-2.1-3.tar.bz2 --no-check-certificate -P ${FRAMEWORK_DIR}
     tar xvfj framework/CUnit-2.1-3.tar.bz2 -C ${FRAMEWORK_DIR}
     cp ${FRAMEWORK_DIR}/CUnit-2.1-3/CUnit/Headers/CUnit.h.in ${FRAMEWORK_DIR}/CUnit-2.1-3/CUnit/Headers/CUnit.h
     echo "Patching Framework"
     cd ${FRAMEWORK_DIR}
-    cp ../src/cunit/cunit_lgpl/patches/CorrectBuildWarningsInCunit.patch  .
+    cp ../src/c_source/cunit_lgpl/patches/CorrectBuildWarningsInCunit.patch  .
     patch -u CUnit-2.1-3/CUnit/Sources/Framework/TestRun.c -i CorrectBuildWarningsInCunit.patch
     echo "Patching Complete"
+else
+    mkdir -p ${FRAMEWORK_DIR}
```

Comment:
shouldn't this just happen no matter what? either if CUnit is there or not... in the C Version doesn't look like your creating the directory for Framework_DIR.